server/src/lib/mutex.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+#include <fty/expected.h>
+#include <optional>
+#include <atomic>
+#include <mutex>
+#include <condition_variable>
+
+namespace fty::storage {
+
+class Mutex
+{
+public:
+    Mutex();
```

Comment:
In case of empty ctor(if no overrides) or dtor (if not virtual), just do not write it.